tabular/src/autogluon/tabular/models/rf/compilers/onnx.py

```diff
@@ -0,0 +1,80 @@
+import os
+import numpy as np
+
+class RFOnnxPredictor:
+    def __init__(self, obj, model):
+        self.num_classes = obj.num_classes
+        self.model = model
+
+    def predict(self, X):
+        input_name = self.model.get_inputs()[0].name
+        label_name = self.model.get_outputs()[0].name
+        return self.model.run([label_name], {input_name: X})[0].squeeze()
+
+    def predict_proba(self, X):
+        input_name = self.model.get_inputs()[0].name
+        label_name = self.model.get_outputs()[1].name
+        pred_proba = self.model.run([label_name], {input_name: X})[0]
+        pred_proba = np.array([[r[i] for i in range(self.num_classes)] for r in pred_proba])
+        return pred_proba
+
+
+class InferenceSessionWrapper:
+    """
+    Wrap around InferenceSession in onnxruntime, since it cannot be pickled.
+    See https://github.com/microsoft/onnxruntime/issues/10097
+    """
+    def __init__(self, onnx_bytes):
+        import onnxruntime as rt
+        self.sess = rt.InferenceSession(onnx_bytes.SerializeToString(),
+                                        providers=['CPUExecutionProvider'])
+
+    def run(self, *args):
+        return self.sess.run(*args)
+
+    def get_inputs(self, *args):
+        return self.sess.get_inputs(*args)
+
+    def get_outputs(self, *args):
+        return self.sess.get_outputs(*args)
+
+    def __getstate__(self):
+        # No need to duplicate the model parameters here.
+        return {}
+
+    def __setstate__(self, values):
+        pass
+
+
+class RFOnnxCompiler:
+    name = 'onnx'
+    save_in_pkl = False
+
+    @staticmethod
+    def can_compile():
+        try:
+            import skl2onnx
+            return True
+        except ImportError:
+            return False
+
+    @staticmethod
+    def compile(obj, path: str, batch_size=None):
```

Comment:
batch_size unused?

Reply:
batch_size not available in abstract model, will cause crashes?

Reply:
> batch_size unused?

batch_size is not used for onnx backend, but would be used for other backends in the future.

> batch_size not available in abstract model, will cause crashes?

Will not crash. batch_size is now available in abstract model. `batch_size` is essentially controlling the compiler to optimize towards either _online_ or _batch_ prediction.